src/Material/Options.elm

```diff
@@ -347,3 +375,72 @@ For example `Textfield`:
 inner : List (Property c m) -> Property { a | inner : List (Property c m) } m
 inner options =
   set (\c -> { c | inner = options ++ c.inner })
+
+
+{-| Add custom event handlers
+ -}
+on : String -> (Decoder.Decoder m) -> Property c m
+on event =
+  Listener event Nothing
+
+
+{-| Add a custom event handler that always succeeds.
+
+Roughly equivalent to `Options.on event (Json.Decode.succeed msg)`
+ -}
+on1 : String -> m -> Property c m
+on1 event m =
+  on event (Decoder.succeed m)
+
+
+{-| Add custom event handlers with options
+ -}
+onWithOptions : String -> Html.Events.Options -> (Decoder.Decoder m) -> Property c m
+onWithOptions evt options =
+  Listener evt (Just options)
+
+
+{-| Add a lifting function that is **required** for multi event dispatch.
+
+**NOTE** If this is missing no events are dispatched with `Options.on`
```

Comment:
Is this true? I'd have thought you just don't get dispatch of multiple handlers for the same event.